yarn-project/archiver/src/archiver/archiver.ts

```diff
@@ -245,8 +251,27 @@ export class Archiver implements ArchiveSource {
     // ********** Events that are processed per L2 block **********
 
     // Read all data from chain and then write to our stores at the end
+    console.log('Rollup Address', this.rollupAddress);
+    console.log('Availability Oracle Address', this.availabilityOracleAddress);
+
     const nextExpectedL2BlockNum = BigInt((await this.store.getBlockNumber()) + 1);
-    const retrievedBlocks = await retrieveBlocks(
+
+    const retrievedBlockBodies = await retrieveBlockBodiesFromDataAvailability(
```

Comment:
Ideas matches what we were talking about yes. Storing the bodies separately for easy access.